### ISSUE TITLE
Discrepancy between PK-Sim and R in applying individual to simulation

### ISSUE THREAD

**Issue by moPellowe** [Open-Systems-Pharmacology/OSPSuite-R#352]
I created an individual with createIndividual() and then applied each of individual$distributedParameters individually to my simulation. When I compare this modified R simulation to an "equivalent" PK-Sim simulation. There is a discrepancy in the simulation results.
![image](https://user-images.githubusercontent.com/57540473/77963049-b121a380-72aa-11ea-9298-afc9af12eb5b.png)
I did a similar test with an IV formulation, but there didn't seem to be an issue there. So perhaps it has to do with the fact that I am using a table formulation?
![image](https://user-images.githubusercontent.com/57540473/77963199-f2b24e80-72aa-11ea-9740-d0ebeca78a1f.png)

I wrote a script to compare the parameters listed in the structure text file exported from PK-Sim and the parameters of a simulation. There are discrepancies between some of the parameters, but it's not clear to me whether the differing parameter values are the reason for the discrepancy.
[diff_values.xlsx](https://github.com/Open-Systems-Pharmacology/OSPSuite-R/files/4405611/diff_values.xlsx)

@msevestre I will send you the R scripts and PK-Sim file via email.

**Comment by moPellowe** [comment]
Here are the files I used to test. There are 2 tests: one for IV and one for the table formulation.
[applying_individual_test.zip](https://github.com/Open-Systems-Pharmacology/OSPSuite-R/files/4405764/applying_individual_test.zip)

**Comment by msevestre** [comment]
@Yuri05 @PavelBal Any idea?

**Comment by msevestre** [comment]
Looking at the table, there are a few parameters with a massive difference


Path | PK-Sim | R | Difference | Relative difference (wrt PK-Sim)
-- | -- | -- | -- | --
Organism\|Age | 40 | 30 | 10 | 0.25
Organism\|Lumen\|Rectum\|Absorption   of liquid | 0.000177 | 0.000145 | 3.19E-05 | 0.180517036
Organism\|BMI | 0.214844 | 0.177557 | 0.037286932 | 0.173553719
Organism\|Lumen\|Duodenum\|Absorption   of liquid | -0.79359 | -0.88172 | 0.088121818 | -0.111041321


Because BMI is different, it means that the height is different. The height is used to scale a bunch of parameters

**Comment by msevestre** [comment]
@Yuri05 What is the `Organism|Lumen|Duodenum|Absorption of liquid` parameter?

**Comment by msevestre** [comment]
@Yuri05 height is not a distributed parameter and as such is not returned as part of the createIndividual algo. I think this is wrong

Now, is height difference enough to explain the discrepancies? Is height used as parameter in  the gi track somewhere, either directly or indirectly?

**Comment by Yuri05** [comment]
>height is not a distributed parameter and as such is not returned as part of the createIndividual algo. I think this is wrong

Agree. Should be returned.

>Now, is height difference enough to explain the discrepancies? Is height used as parameter in the gi track somewhere, either directly or indirectly?

Yes, GI-geometry (length, proximal/distal radius) is height dependent

**Comment by Yuri05** [comment]
@msevestre What I also don't understand in your table above: why is there a difference in the Age?!

**Comment by PavelBal** [comment]
This is exactly what I was talking about in https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/319

`Age` is part of the "derived" parameters, therefore are not applied (because should not be, according to documentation) in R.

The method should return the distributed, the derived (formula-defined), and the "constant" parameters.

**Comment by msevestre** [comment]
I am not sure what you mean @PavelBal 
Age is a distributed parameter and should be returned.
Height is not and this is the problem


>The method should return the distributed, the derived (formula-defined), and the "constant" parameters.

I disagree with this. You do not need those parameters are they are NOT updated when you change age. In fact, if you return everything, what's the point of formula? You will override all of them with constant values

The issue is between species scaling. E.g. starting with a mouse and setting human values. In that case we would need the constant but this is something we cannot use until the model structure is set..

@Yuri05 the age here is just a script error. PKSim creates 40 years old but the script 30 (or vice versa). This should make absolutely no difference in the output in this age range (or very minimal)

**Comment by msevestre** [comment]
@PavelBal You are correct. Age is in the derived list and this is wrong
I will move, Age, Height, GA etc.. to the distributed list

**Comment by msevestre** [comment]
I have identified following parameters that should be moved from one list to the other
Age, 
Height,
GestationalAge
Ontogeny factor (alpha1-acid glycoprotein)
Ontogeny factor (albumin)

Also I will remove MeanBW and MeanHeight that are internal parameters of PKSim

**Comment by msevestre** [comment]
@moPellowe Can you try setting the height explictely in your simulation AFTER updating the parameters and see if this solve the problem?

**Comment by moPellowe** [comment]
That seems to work. 👍 
![image](https://user-images.githubusercontent.com/57540473/78046494-39548700-7345-11ea-9057-ef5b26dc06d8.png)

**Comment by msevestre** [comment]
cool. Ok new R and PKSim coming today :)

Thanks for your perseverance here @moPellowe

**Comment by msevestre** [comment]
This will be fixed in the next version. Latest version of PKSim portable required

**Comment by moPellowe** [comment]
I ended up finding the discrepancy again even after the fix. I redid the test but with a more lipophilic drug (more similar to the drug I'm working with) and the R simulation doesn't match the PK-Sim simulation again. The script I ran shows that there are still slight discrepancies in some parameters. Based on the plot, it looks like there are differences in clearance that are not being captured in the R simulation.
![image](https://user-images.githubusercontent.com/57540473/78177392-0fc25b00-742c-11ea-9983-f1cd4f023df5.png)
[diff_values.xlsx](https://github.com/Open-Systems-Pharmacology/OSPSuite-R/files/4417156/diff_values.xlsx)
[S-Warfarin_test.zip](https://github.com/Open-Systems-Pharmacology/OSPSuite-R/files/4417169/S-Warfarin_test.zip)

**Comment by msevestre** [comment]
Did you update PK-Sim as well?

**Comment by msevestre** [comment]
BEcause it looks like setting the height by hand works. So that means that height is not part of the parameter retrieved automatically. Therefore it seems that something is not quite updated.

Please let me know

**Comment by moPellowe** [comment]
I'm using PK-Sim 9.0.119, which I believe is the latest version. The plot also shows that setting the height manually doesn't fix the issue.

**Comment by msevestre** [comment]
Oh I see. I minsterpreted the legend. 
Well let's check it again :)

**Comment by msevestre** [comment]
@moPellowe Looking at the code, you can simply do 

```
setParameterValuesByPath(individual$distributedParameters$paths, individual$distributedParameters$values, simulation = sim)
```


Note also that values returned by createIndividuals etc.. are always in BASE UNIT so you do not need to convert

@PavelBal Probably worth adding to the documentation too

This is not the problem however :)

**Comment by msevestre** [comment]
ok the problem I believe is in the fact that one simulation was created using an individual with enzymes 

![image](https://user-images.githubusercontent.com/1041237/78187218-093bdf80-743c-11ea-9ca9-f3ea66107e3b.png)

You use this individual to create the simulation
![image](https://user-images.githubusercontent.com/1041237/78187419-5a4bd380-743c-11ea-8d89-8a8d44ee5460.png)


However the one you are comparing too is using another individual
![image](https://user-images.githubusercontent.com/1041237/78187299-2a043500-743c-11ea-82ab-13e703b9a0f0.png)

![image](https://user-images.githubusercontent.com/1041237/78187374-486a3080-743c-11ea-8e5c-e568eac7ef3b.png)

Therefore the simulation WILL NOT HAVE THE ENZYMES => no clearance process involved

So basically you are comparing a simulation with enzymatic process with one without. This is not an issue with the createIndivdual algorithm. It only creates values for you but does not change the simulation structure.

**Comment by msevestre** [comment]
If you want to verify that the create individual is working, you need to create use an individual without enzymes or you need to add the same enzyme to your Individual-Asian female.

Let me know if this is not clear.
